src/main.cpp

```diff
@@ -2921,14 +2921,8 @@ bool CheckBlock(
         is_enum_any_of(state.getTxOrigin(), TxOrigin::MINED_BLOCK, TxOrigin::MSG_BLOCK, TxOrigin::GENERATED))
     {
         // signature validation of the previous block's merkle root is already done in CheckBlockHeader
-        // so here we need to check that PastelID (mnid) of the signer is registered
-        masternode_info_t mnInfo;
-        if (!masterNodeCtrl.masternodeManager.GetAndCacheMasternodeInfo(block.sPastelID, mnInfo))
-        {
-            strRejectReasonDetails = strprintf("MasterNode with mnid='%s' is not in the list", block.sPastelID);
-            return state.DoS(10, error("%s: %s", __func__, strRejectReasonDetails),
-                REJECT_INVALID, "mn-not-listed", false, strRejectReasonDetails);
-        }
+        // so here we need to check that PastelID (mnid) of the signer was eligible to mine the block
+        // mnid ticket is already checked above

```

Comment:
@akobrin1 wait, you just remove the check altogether?

Reply:
yes, I already have a check that mind ticket exists in that function